### PR TITLE
Fixed the students locating logic

### LINKED ISSUE #19: Problem with current get_students
The function "get_students" returns the IDs of students. If the students_names parameters received, then the IDs of the students according to the names received. Otherwise, the IDs of all of the students enrolled to the course are returned.
There is a problem with the logic in this function:
The current state of this issue is that the logic in the Moodler will not find any student with the name received but will only find the first one. For instance:
Let's say the list of the students received by the user is ['aqil', ... ].
Let's say there are two students in the Moodle with that name: "Aqil Bla", "Aqil Sba". The current logic will only find the first aqil.

### PR DESCRIPTION
**Pull request by almogcarmi**

### CODE REVIEW

File: moodler/students.py [hunk 1]
```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
...
+                logger.error("Already found student - skipping this name")
+                return False
+
+            students_names_dict[student_name] = True
 
     return False
 
 
-def get_students(course_id, students_names=None):
+def get_students(course_id, students_names: list = None):
```
Comment (maxandron) [inline moodler/students.py]: I'm having trouble understanding this function in this context...
Document the parameters please. What is student_names? When would this be None?
Also, students_names_dict is basically a dict of student_name to found_status? If yes, change the name to be more descriptive, maybe add a comment. students_names_dict is not informative enough...

File: moodler/students.py [hunk 2]
```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
     return response
 
 
-def is_student_in_names(students_names, enrolled):
+def is_student_in_names(students_names_dict, enrolled):
```
Comment (maxandron) [inline moodler/students.py]: So this function always returns False, no?

File: moodler/students.py [hunk 3]
```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
...
+        students_names_dict = dict.fromkeys(students_names, False)
+
     for enrolled in core_enrol_get_enrolled_users(course_id):
         if enrolled['roles'][0]['shortname'] != 'student':
             continue
 
-        if students_names is not None:
-            if not is_student_in_names(students_names, enrolled):
+        if students_names_dict is not None:
+            if not is_student_in_names(students_names_dict, enrolled):
```
Comment (maxandron) [inline moodler/students.py]: Wouldn't this always continue?

**Review by maxandron — CHANGES_REQUESTED** [review]

File: moodler/students.py (hunk 1 above)
Comment (almogcarmi) [reply moodler/students.py]: You are totally right. I have decided that I was trying to enforce two functions to be one function.
I created a whole new function to be used for retrieving students IDs by their names.

File: moodler/sections.py [hunk 4]
```diff
@@ -75,13 +75,12 @@ def list_courses(course_prefix=None):
 
     for course in courses_from_moodle:
         if course_prefix is not None:
-            if ((course_prefix not in course['shortname']) and
-                    (course_prefix not in course['fullname'])):
+            if (course_prefix not in course["shortname"]) and (
```
Comment (maxandron) [inline moodler/sections.py]:
```python
if not (course_prefix in course["shortname"] or course_prefix in course["full_name"])
```

Comment (maxandron) [reply moodler/sections.py]: actually something better

```python
if course_prefix not in (course["shortname"] + course["full_name"])
```

Comment (maxandron) [reply moodler/sections.py]: Unless `.starts_with` is more appropriate here

File: moodler/sections.py [hunk 5]
```diff
@@ -134,76 +138,64 @@ def locate_course_name(course_id, course_prefix=None):
...
 
-            sections_not_found.remove(section['name'])
-
-        current_section_assignments = []
-        for module in section['modules']:
-            # We are only looking for assignments and no other resource in
-            # the moodle
-            if module['modname'] != 'assign':
+            # Filter assignments by name
+            if assignments_names and not module["name"] in assignments_names:
```
Comment (maxandron) [inline moodler/sections.py]: `not "a" in "abc"`
is the same as `"a" not in "abc"`

**Review by maxandron — CHANGES_REQUESTED** [review]

**Pull request merged**